doc/source/working_with_kiwi.rst

```diff
@@ -84,3 +85,148 @@ to find in the *description directory*:
+following package managers: ``dnf``, ``zypper`` (default) and ``apt-get``.
+
+The prepare step consists of the following substeps:
+
+#. **Create Target Root Directory**
+
+   KIWI aborts with an error if the target root tree already exists to
+   avoid accidental deletion of an existing unpacked image.
+
+#. **Install Packages**
+
+   First, KIWI configures the package manager to use the repositories
+   specified in the configuration file and/or via the command line. After
+   the repository setup, the packages specified in the ``bootstrap``
+   section of the image description are installed in a temporary directory
+   external to the target root tree. This establishes the initial
+   environment to support the completion of the process in a chroot
+   setting. The essential packages are ``filesystem`` and ``glibc-locale`` to specify
+   as part of the bootstrap. The
+   dependency chain of these two packages is usually sufficient to populate
+   the bootstrap environment with all required software to support the
+   installation of packages into the new root tree. The
+   aforementioned two packages might not be enough for every distribution.
+   Consult the `kiwi-descriptions repository
+   <https://github.com/SUSE/kiwi-descriptions/>`_ for examples for various
+   Linux distributions.
+
+   The installation of software packages through the selected package
+   manager may install unwanted packages. Removing these packages can be
+   accomplished by marking them for deletion in the image description, see
+   :ref:`xml-description-adding-and-removing-packages`.
+
+#. **Apply the Overlay Tree**
+
+   Next, KIWI applies all files and directories present in the overlay
+   directory named :file:`root` or in the compressed overlay
+   :file:`root.tar.gz` to the target root tree. Files already present in
+   the target root directory are overwritten. This allows you to
+   overwrite any file that was installed by one of the packages during the
+   installation phase.
+
+#. **Apply Archives**
+
+   All archives specified in the `archive` element of the
+   :file:`config.xml` file are applied in the specified order (top to
+   bottom) after the overlay tree copy operation is complete (see
+   :ref:`xml-description-archive-element`). Files and directories are
+   extracted relative to the top level of the new root tree. As with the
+   overlay tree, it is possible to overwrite files already existing in the
+   target root tree.
+
+#. **Execute the user-defined script** :file:`config.sh`
+
+   At the end of the preparation stage the script :file:`config.sh` is
+   executed (if present). It is run in the top level directory of the
+   target root tree. The script's primary function is to complete the
+   system configuration, for example, to activate services. See
+   :ref:`image-customization-config-sh` section for further details.
+
+#. **Manage The New Root Tree**
+
+   .. FIXME: I don't understand the purpose of this paragraph, is this even
```

Comment:
@schaefi @davidcassany I stumbled over this paragraph and included it more or less unchanged, but I don't really get it's meaning and purpose.

Reply:
kiwi operates in two steps. The prepare step that creates a new root tree and the create step that creates an image using that tree. With the section "Manage The New Root Tree" we wanted to point out that customers can do manual changes in the root tree after the prepare command is done. kiwi's build command basically combines the two commands prepare + create

So there is the possibility to call kiwi like this:
   
kiwi system prepare ... [plugin-customer-changes] ... kiwi system create ...

Usually customers pack the custom changes into a config.sh script and use kiwi system build

Reply:
I've rephrased this paragraph a bit and moved the warning into a `.. warning::` box.

I am however a little puzzled about the warning: the whole text says, yeah you can now modify the unpacked image if you want to and then the warning says: "Don't you dare do this!" (both sides exaggerated by me). Maybe we should rephrase this, so that the text doesn't carry this odd ambivalence?